bootstrap/092_service_account_setup.sql

```diff
@@ -1,15 +1,59 @@
 
+-- allows a custom API Integration to be used instead of defaulting to OPSCENTER_SUNDECK_EXTERNAL_FUNCTIONS
+CREATE OR REPLACE PROCEDURE admin.finalize_setup_from_service_account(integration_name varchar, url varchar, web_url varchar)
+RETURNS varchar
+LANGUAGE sql
+as
+begin
+    execute immediate 'create or replace function internal.get_ef_url() returns string as \'\\\'' || url || '\\\'\';';
+    execute immediate 'create or replace function internal.get_tenant_url() returns string as \'\\\'' || web_url || '\\\'\';';
+
+    -- Create the task so we can run finalize_setup asynchronously
+    CREATE OR REPLACE TASK TASKS.UPGRADE_CHECK
+        SCHEDULE = '1440 minute'
+        ALLOW_OVERLAPPING_EXECUTION = FALSE
+        USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "XSMALL"
+        AS
+        CALL ADMIN.UPGRADE_CHECK();
+
+    call internal.set_config('tenant_url', :web_url);
+    call internal.set_config('url', :url);
+
+    let ref_name varchar := (select 'OPSCENTER_API_INTEGRATION');
+    let ref_val varchar := (select SYSTEM$REFERENCE('API Integration', :integration_name, 'persistent', 'usage'));
+
+    // We can't use admin.update_reference because it expects that the OpsCenter token is already set.
+    select system$set_reference(:ref_name, :ref_val);
+    insert into internal.reference_management (ref_name, operation, ref_or_alias) values (:ref_name, 'finalize_setup_from_service_account', :ref_val);
+end;
+
 CREATE OR REPLACE PROCEDURE admin.finalize_setup_from_service_account(url varchar, web_url varchar)
 RETURNS varchar
 LANGUAGE sql
 as
+declare
+    integration_name text default (select 'OPSCENTER_SUNDECK_EXTERNAL_FUNCTIONS');
```

Comment:
I don't think this is the right name. That doesn't match the manifest or the api integration we suggest creating in the help script